tests/test_hra.py

```diff
@@ -982,6 +911,147 @@ def test_exception_invalid_decay(self):
         self.assertIn('Invalid decay type bad decay type provided',
                       str(cm.exception))
 
+    def test_summary_stats(self):
+        """HRA: test summary stats table."""
+        from natcap.invest import hra
+        e_array = numpy.array([[0, 1, 2, 3]], dtype=numpy.float32)
+        c_array = numpy.array([[0.5, 1.5, 2.5, 3.5]], dtype=numpy.float32)
+        risk_array = numpy.array([[0, 1.1, 2.2, 3.3]], dtype=numpy.float32)
+        classes_array = numpy.array([[0, 1, 2, 3]], dtype=numpy.int8)
+
+        pairwise_raster_dicts = [{
+            'habitat': 'life',
+            'stressor': 'industry',
+            'e_path': os.path.join(self.workspace_dir, 'e.tif'),
+            'c_path': os.path.join(self.workspace_dir, 'c.tif'),
+            'risk_path': os.path.join(self.workspace_dir, 'risk.tif'),
+            'classification_path': os.path.join(self.workspace_dir,
+                                                'classes.tif'),
+        }]
+        nodata = -1
+        for array, key in [
+                (e_array, 'e_path'),
+                (c_array, 'c_path'),
+                (risk_array, 'risk_path'),
+                (classes_array, 'classification_path')]:
+            pygeoprocessing.numpy_array_to_raster(
+                array, nodata, (10, -10), ORIGIN, SRS_WKT,
+                pairwise_raster_dicts[0][key])
+
+        # For the sake of testing this function more rigorously, creating a new
+        # classification path for the per-habitat summary classification
+        # raster.
+        per_habitat_classifications = {
+            pairwise_raster_dicts[0]['habitat']: os.path.join(
+                self.workspace_dir, 'summary_classes.tif')
+        }
+        pygeoprocessing.numpy_array_to_raster(
+            numpy.array([[2, 3, 2, 3]], dtype=numpy.int8),
```

Comment:
Are these values in the array directly related to the expected values of the summary stats? If so, is there a way -- either a variable or just a comment -- to make that connection?

Reply:
Yes, absolutely.  In edaa069f7 I reworked the test so that variable names could be more clear and also to derive test values from the arrays they're generated from.  I hope that'll help make the connection ... I think this should make the test easier to maintain too, but let me know what you think!

Reply:
Thanks that's very helpful!